widget/view.controller.js

```diff
@@ -1,72 +1,121 @@
-                        $scope.showSpinner = _.has(foundPickListValue, 'showSpinner')  ? foundPickListValue.showSpinner : true;
-                        $scope.showIcon = foundPickListValue.showIcon
-                        $scope.loading = true;
-                    } else  {
-                        $scope.title = "";
-                        $scope.picklistMessageIcon = "";
-                        $scope.showSpinner = false;
-                        $scope.showIcon = false;
-                        $scope.loading = false;
-                    }
-                });
-         }
+       .module('cybersponse')
+       .controller('customPicklistMessageDev200Ctrl', customPicklistMessageDev200Ctrl);
+    customPicklistMessageDev200Ctrl.$inject = ['$scope', '$state', '$timeout', '$interval', 'Modules', 'Entity', 'config', 'websocketService', '$rootScope', '_'];
+ 
+    function customPicklistMessageDev200Ctrl($scope, $state, $timeout, $interval, Modules, Entity, config, websocketService, $rootScope, _) {
+       $scope.config = config;
+       $scope.title = '';
+       $scope.timeinterval = 0;
+       $scope.currentTheme = $rootScope.theme.id;
+       var widgetsubscription;
+ 
+       function checkPicklistCondition() {
+          Modules.get({
+             module: $state.params.module,
+             id: $state.params.id,
+             __selectFields: config.picklistItem
+          }).$promise.then(function (result) {
+             var foundPickListValue = _.find($scope.config.picklits_mapping.options, function (item) {
+                return item.picklistValue === result[config.picklistItem].itemValue;
+             });
+             if (foundPickListValue) {
+                $scope.flag = true
+                $scope.title = foundPickListValue.picklistLoadingTitle;
+                $scope.picklistMessageIcon = foundPickListValue.picklistMessageIcon;
+                $scope.showSpinner = _.has(foundPickListValue, 'showSpinner') ? foundPickListValue.showSpinner : true;
+                $scope.showIcon = foundPickListValue.showIcon
+                $scope.loading = true;
+                if (config.time) {
+                    $timeout(function () {
+                       $scope.flag = false
+                       $scope.title = config.newtitle;
+                       if (config.newpicklistItem && config.newpicklistValue) {
+                         updatePicklist();
+                       }
+                    }, config.time*60000);
+                 }
+             } else {
+                $scope.title = "";
+                $scope.picklistMessageIcon = "";
+                $scope.showSpinner = false;
+                $scope.showIcon = false;
+                $scope.loading = false;
+             }
+          })
+       }
+ 
+        function updatePicklist() {
+          var picklistObject = {};
+          picklistObject[config.newpicklistItem] = $scope.selectedPicklistObject
```

Comment:
$scope.config.picklistItem will be used. please verify

Reply:
updated